pandas/core/frame.py

```diff
@@ -1399,11 +1399,45 @@ def to_dict(self, orient="dict", into=dict):
             )
         # GH16122
         into_c = com.standardize_mapping(into)
-        if orient.lower().startswith("d"):
+        orient = orient.lower()
```

Comment:
Should remove this; caller should be responsible for correct case

Reply:
Agreed, I had removed it in my first PR, but added it back in to open a new issue to remove this in another PR (see discussion from previous PRs)

Reply:
Ah OK I see

Reply:
Though I think you should check the orient before dropping to lowercase. Right now this would still accept orients like SPLIT which I don’t think it should. 

If you can parametrize your test and cover that deprecation would be good as well

Reply:
yes I was thinking about this too, but other methods like `from_dict()` contain orient.lower() too, so I was gonna deal with that separately to keep this small. I'd rather deal with all the unnecessary .lower() at the same time. What do you think?

Reply:
(this is my first time contributing open source, and i would like to get this right, so superthankful for all these comments, btw!)

Reply:
Sorry again...yea I misunderstood the thread but looks like this topic was already covered. Thanks for the insights and great job on this so far!